src/hotspot/share/runtime/thread.hpp

```diff
@@ -1172,13 +1173,17 @@ class JavaThread: public Thread {
   bool on_thread_list() const { return _on_thread_list; }
   void set_on_thread_list() { _on_thread_list = true; }
 
-  // thread has called JavaThread::exit() or is terminated
+  // thread has called JavaThread::exit(), thread's GC barrier is detached
+  // or thread is terminated
   bool is_exiting() const;
+  // thread's GC barrier is detached or thread is terminated
+  bool is_gc_barrier_detached() const;
```

Comment:
We can definitely rename, but naming is hard... :-)

Those names you proposed are inversions of the current name and would
require that the logic of the function be inverted.

The other names I came up before I settled on is_gc_barrier_detached():

```
    can_no_longer_access_oops()
    oops_are_no_longer_safe()
    cannot_safely_access_oops()
```

The closest match between your proposed names and mine are:

`can_access_oops_safely`  and `cannot_safely_access_oops`

Other opinions are welcome.

Reply:
I think the style guide for naming is to use positive names to avoid double negatives when you have to negate the function - so `is_safe_for_x` rather than `is_not_safe_for_x`. I like `thread->is_oop_safe()` - short and clear.

Reply:
I do one more rename of the bug, the PR and the function name...